client/external-declarations.d.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * External dependencies
+ */
+import '@wordpress/date';
+
+declare module '@wordpress/date' {
+	export function dateI18n(
+		dateFormat: string,
+		dateValue: import('moment').Moment | Date | string | undefined,
+		timezone?: string | boolean
```

Comment:
I redefined this function with an optional `timezone` parameter instead of `timezone: string | boolean | undefined` like the original built-in type.

This avoids `tsc` complaining about a missing 3rd argument `timezone`, which is in practice optional, yet not typed as such in the built-in types.